internal/discovery/namespace_filter.go

```diff
@@ -0,0 +1,126 @@
+package discovery
+
+import (
+	"errors"
+
+	"github.com/newrelic/nri-kubernetes/v3/internal/config"
+	"github.com/newrelic/nri-kubernetes/v3/internal/storer"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	listersv1 "k8s.io/client-go/listers/core/v1"
+)
+
+// NamespaceFilterer provides an interface to filter namespaces.
+type NamespaceFilterer interface {
+	IsAllowed(namespace string) bool
+}
+
+// NamespaceFilter is a struct holding pointers to the config and the namespace lister.
+type NamespaceFilter struct {
+	c      *config.Config
+	lister listersv1.NamespaceLister
+	stopCh chan<- struct{}
+	storer storer.Storer
+}
+
+// NewNamespaceFilter inits the namespace lister and returns a new NamespaceFilter.
+func NewNamespaceFilter(c *config.Config, client kubernetes.Interface, storer storer.Storer, options ...informers.SharedInformerOption) *NamespaceFilter {
```

Comment:
Just to check if I'm understanding it correctly: in worst case scenario, something is changed and the integration won't notice until cache's ttl elapses, right?

Reply:
> Yes, I thought in setting a properly TTL and interval to fit within each integration run. WDYT?

I think it is OK, I asked because we have two different caches: one cache is handled internally by the namespace informer, an another handled by the storer. I wonder if the storer TTL and the informer resync time should be related.

Reply:
I added the cache to clean itself on each interval. This interval is configurable and should be smaller than the integration interval. Done in 89b8e6004d8f746a9c2ce9cdaa28c96d7ca696e4